src/casts/specify-representation.rkt

```diff
@@ -479,53 +490,42 @@ but a static single assignment is implicitly maintained.
-               crcn COERCION-TUPLE-COUNT-INDEX tagged-count)
-              (sr-tag-value crcn COERCION-MEDIATING-TAG))
-            (begin$
-              (assign$ c1a
-                (sr-tagged-array-ref
-                 crcn1
-                 COERCION-MEDIATING-TAG
-                 (sr-plus COERCION-TUPLE-ELEMENTS-OFFSET i)))
-              (assign$ c2a
-                (sr-tagged-array-ref
-                 crcn2
-                 COERCION-MEDIATING-TAG
-                 (sr-plus COERCION-TUPLE-ELEMENTS-OFFSET i)))
-              (assign$ composed-crcn (App-Code comp-crcn-label `(,c1a ,c2a)))
-              (assign$ id?2
-                (If id?1
-                    (sr-check-tag=?
-                     composed-crcn COERCION-TAG-MASK COERCION-IDENTITY-TAG)
-                    FALSE-IMDT))
-              (assign$ tmp-crcn
-                (App-Code
-                 comp-tuple-crcn-label
-                 `(,crcn1 ,crcn2 ,(sr-plus (Quote 1) i) ,count ,id?2)))
-              (If (sr-check-tag=?
-                   tmp-crcn COERCION-TAG-MASK COERCION-IDENTITY-TAG)
-                  COERCION-IDENTITY-IMDT
-                  (begin$
-                    (sr-tagged-array-set!
-                     tmp-crcn
-                     COERCION-MEDIATING-TAG
-                     (sr-plus COERCION-TUPLE-ELEMENTS-OFFSET i)
-                     composed-crcn)
-                    tmp-crcn))))))
+      (code$ (crcn1 crcn2)
+        (assign$ tagged-count
+          (sr-tagged-array-ref
+           crcn1 COERCION-MEDIATING-TAG COERCION-TUPLE-COUNT-INDEX))
+        (assign$ count (op$ %>> tagged-count COERCION-SECOND-TAG-SHIFT))
+        (assign$ new-crcn UNIT-IMDT)
+        (assign$ iters (op$ + COERCION-TUPLE-ELEMENTS-OFFSET count))
+        (assign$ id? TRUE-IMDT)
+        (repeat$ (i COERCION-TUPLE-ELEMENTS-OFFSET iters) (_ UNIT-IMDT)
+          (assign$ c1a (sr-tagged-array-ref crcn1 COERCION-MEDIATING-TAG i))
+          (assign$ c2a (sr-tagged-array-ref crcn2 COERCION-MEDIATING-TAG i))
+          (assign$ composed-crcn (app-code$ comp-crcn-label c1a c2a))
+          (cond$
+           [id?
+            (assign$ tmp-id?
+              (sr-check-tag=?
+               composed-crcn COERCION-TAG-MASK COERCION-IDENTITY-TAG))
+            (cond$
+             [(op$ not tmp-id?)
+              (Assign (Var-id new-crcn)
+                (op$ Alloc (op$ + count COERCION-TUPLE-ELEMENTS-OFFSET)))
+              (sr-array-set! new-crcn COERCION-TUPLE-COUNT-INDEX tagged-count)
+              (repeat$ (j COERCION-TUPLE-ELEMENTS-OFFSET i) (_ UNIT-IMDT)
+                (sr-array-set! new-crcn j COERCION-IDENTITY-IMDT))
+              (sr-array-set! new-crcn i composed-crcn)
+              (Assign (Var-id id?) FALSE-IMDT)]
+             [else UNIT-IMDT])]
+           [else (sr-array-set! new-crcn i composed-crcn)]))
+        (cond$
+         [id? COERCION-IDENTITY-IMDT]
+         [else (sr-tag-value new-crcn COERCION-MEDIATING-TAG)])))
```

Comment:
This needs some documentation.

Reply:
Something like "this checks if the result coercion is an identity so that allocation can be avoided."?

Reply:
I was referring to the entire code$ form here.

Reply:
Done.